File: src/providers/xxx.py
"""预留 provider（占位实现）。"""

from __future__ import annotations

from typing import Any, Dict, Optional


def invoke_xxx(
    prompt: str,
    session_id: Optional[str] = None,
    stream: bool = True,
    *,
    timeout_level: str = "standard",
    idle_timeout_s: Optional[float] = None,
    max_timeout_s: Optional[float] = None,
    terminate_grace_s: Optional[float] = None,
) -> Dict[str, Any]:
    """调用占位 provider。

    说明：
    - 当前不连接真实模型
    - 仅回显输入，便于先跑通统一接口
    """
    # 预留参数：后续真实接入时可用于会话恢复。
    _ = session_id
    _ = timeout_level
    _ = idle_timeout_s
    _ = max_timeout_s
    _ = terminate_grace_s

    text = f"[xxx placeholder] prompt received: {prompt}"
    if stream:
        # 模拟流式输出：这里直接一次性打印文本。
        print(text)

    return {
        "provider": "xxx",
        "text": text,
        "session_id": None,
    }

File: src/providers/xxx.js
"use strict";

// 这是“预留 provider”示例：
// 现在不真正调用外部 AI，只返回固定说明，方便统一接口先跑通。
async function invokeXxx(options) {
  const { prompt, stream = true } = options;
  const text = `[xxx placeholder] 已收到你的问题：${prompt}`;

  if (stream) {
    process.stdout.write(`${text}\n`);
  }

  return {
    provider: "xxx",
    text,
    sessionId: null,
  };
}

module.exports = {
  invokeXxx,
};

